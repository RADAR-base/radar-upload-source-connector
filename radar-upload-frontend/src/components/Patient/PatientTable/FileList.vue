<template>
  <v-list
    shaped
    two-line
    subheader
  >
    <v-layout justify-center>
      <v-progress-circular
        class="mt-2"
        v-show="loading"
        indeterminate
        color="primary"
      />
      <v-alert
        type="error"
        v-text="error"
        v-show="error"
        dense
        text
      />
    </v-layout>
    <v-subheader>
      Current files
    </v-subheader>

    <v-list-item-group color="primary">
      <v-list-item
        v-for="(file, i) in patientFiles"
        :key="i"
      >
        <v-list-item-icon>
          <v-icon>mdi-music-circle-outline</v-icon>
        </v-list-item-icon>

        <v-list-item-content>
          <v-list-item-title>
            {{ file.fileName }}
          </v-list-item-title>

          <v-list-item-subtitle>
            {{ file.uploadedAt | moment("YYYY/MM/DD") }}
          </v-list-item-subtitle>
        </v-list-item-content>

        <v-list-item-action>
          <v-icon color="success">
            mdi-check-circle
          </v-icon>
        </v-list-item-action>
      </v-list-item>
    </v-list-item-group>
  </v-list>
</template>

<script>
export default {
  props: {
    patientFiles: {
      type: Array,
      required: true,
    },
    error: {
      type: String,
      default: '',
    },
    loading: {
      type: Boolean,
      default: false,
    },
  },
};
</script>

<style>

</style>
