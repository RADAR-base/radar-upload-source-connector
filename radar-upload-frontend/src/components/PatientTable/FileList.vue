<template>
  <v-list
    shaped
    two-line
    subheader
  >
    <v-subheader>
      Current files
    </v-subheader>

    <v-list-item-group color="primary">
      <v-list-item
        v-for="(file, i) in patientFiles"
        :key="i"
      >
        <v-list-item-icon>
          <v-icon>mdi-music-circle-outline</v-icon>
        </v-list-item-icon>

        <v-list-item-content>
          <v-list-item-title>
            {{ file.fileName }}
          </v-list-item-title>

          <v-list-item-subtitle>
            {{ file.uploadedAt | moment("YYYY/MM/DD") }}
          </v-list-item-subtitle>
        </v-list-item-content>

        <v-list-item-action>
          <v-icon color="success">
            mdi-check-circle
          </v-icon>
        </v-list-item-action>
      </v-list-item>
    </v-list-item-group>
  </v-list>
</template>

<script>
export default {
  props: {
    patientFiles: {
      type: Array,
      required: true,
    },
  },
};
</script>

<style>

</style>
