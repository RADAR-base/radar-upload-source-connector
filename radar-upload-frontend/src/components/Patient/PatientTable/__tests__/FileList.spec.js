/* eslint-disable no-undef */
import { shallowMount } from '@vue/test-utils';
import FileList from '../FileList.vue';

// eslint-disable-next-line no-undef
describe('ProjectFilter', () => {
  const files = [
    {
      sequence: 1,
      fileName: 'Audio1',
      fileType: 'mp3',
      status: 'Incomplete',
      uploadedAt: 1565186733673,
    },
  ];
  const wrapper = shallowMount(FileList, {
    propsData: {
      patientFiles: files,
      error: '',
      loading: false,
    },
    stubs: [
      'v-list',
      'v-subheader',
      'v-list-item-group',
      'v-list-item',
      'v-list-item-icon',
      'v-list-item-content',
      'v-list-item-title',
      'v-list-item-subtitle',
      'v-list-item-action',
      'v-icon',
      'v-alert',
      'v-progress-circular',
      'v-layout',
    ],
    filters: {
      moment: () => 'filteredDate',
    },
  });
  it(' get and render props patientFiles correctly', () => {
    expect(wrapper.vm.patientFiles).toEqual(files);
    expect(wrapper.vm.loading).toEqual(false);
    expect(wrapper.vm.error).toEqual('');
    expect(wrapper.text()).toContain(files[0].fileName);
    expect(wrapper.text()).toContain('filteredDate');
  });

  it('show error  and loading if any', () => {
    expect(wrapper.find('v-progress-circular-stub').isVisible()).toBe(false);
    expect(wrapper.find('v-alert-stub').isVisible()).toBe(false);
    wrapper.setProps({ error: 'error', loading: true });
    expect(wrapper.find('v-progress-circular-stub').isVisible()).toBe(true);
    expect(wrapper.find('v-alert-stub').isVisible()).toBe(true);
  });
  // it('match snapShopt', () => {
  //   expect(wrapper.html()).toMatchSnapshot();
  // });
});
