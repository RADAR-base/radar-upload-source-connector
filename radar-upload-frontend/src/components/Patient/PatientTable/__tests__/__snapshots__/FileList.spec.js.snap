// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectFilter match snapShopt 1`] = `
<v-list-stub shaped="" two-line="" subheader="">
  <v-subheader-stub>
    Current files
  </v-subheader-stub>
  <v-list-item-group-stub color="primary">
    <v-list-item-stub>
      <v-list-item-icon-stub>
        <v-icon-stub>mdi-music-circle-outline</v-icon-stub>
      </v-list-item-icon-stub>
      <v-list-item-content-stub>
        <v-list-item-title-stub>
          Audio1
        </v-list-item-title-stub>
        <v-list-item-subtitle-stub>
          filteredDate
        </v-list-item-subtitle-stub>
      </v-list-item-content-stub>
      <v-list-item-action-stub>
        <v-icon-stub color="success">
          mdi-check-circle
        </v-icon-stub>
      </v-list-item-action-stub>
    </v-list-item-stub>
  </v-list-item-group-stub>
</v-list-stub>
`;
